### ISSUE TITLE
💅 `while (true)` is not treated as an infinite loop: `noUnreachable` misses code after it, `useGetterReturn` false positive

### ISSUE THREAD

**Issue by AlbinoGeek** [biomejs/biome#11946]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.14

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:           npm
  JS_RUNTIME_NAME:              node
```

</details>

### Rule name

`correctness/noUnreachable`, `suspicious/useGetterReturn`

### Playground link

https://biomejs.dev/playground/?lintRules=noUnreachable#files.noUnreachable.js=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABhACgAKQAgAHsACgAgACAAZgBvAHIAIAAoADsAOwApACAAewB9AAoAIAAgAHIAZQB0AHUAcgBuACAAMQA7ACAALwAvACAAcgBlAHAAbwByAHQAZQBkAAoAfQAKAGUAeABwAG8AcgB0ACAAZgB1AG4AYwB0AGkAbwBuACAAYgAoACkAIAB7AAoAIAAgAHcAaABpAGwAZQAgACgAdAByAHUAZQApACAAewB9AAoAIAAgAHIAZQB0AHUAcgBuACAAMQA7ACAALwAvACAAbgBvAHQAIAByAGUAcABvAHIAdABlAGQACgB9AAoAZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABjACgAKQAgAHsACgAgACAAZABvACAAewB9ACAAdwBoAGkAbABlACAAKAB0AHIAdQBlACkAOwAKACAAIAByAGUAdAB1AHIAbgAgADEAOwAgAC8ALwAgAG4AbwB0ACAAcgBlAHAAbwByAHQAZQBkAAoAfQAKAA%3D%3D

https://biomejs.dev/playground/?lintRules=useGetterReturn#files.useGetterReturn.js=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABvACAAPQAgAHsACgAgACAAZwBlAHQAIABhACgAKQAgAHsACgAgACAAIAAgAGYAbwByACAAKAA7ADsAKQAgAHsAfQAgAC8ALwAgAG4AbwB0ACAAcgBlAHAAbwByAHQAZQBkAAoAIAAgAH0ALAAKACAAIABnAGUAdAAgAGIAKAApACAAewAKACAAIAAgACAAdwBoAGkAbABlACAAKAB0AHIAdQBlACkAIAB7AH0AIAAvAC8AIAByAGUAcABvAHIAdABlAGQAOgAgACIAVABoAGkAcwAgAGcAZQB0AHQAZQByACAAcwBoAG8AdQBsAGQAIAByAGUAdAB1AHIAbgAgAGEAIAB2AGEAbAB1AGUALgAiAAoAIAAgAH0ALAAKAH0AOwAKAA%3D%3D

```js
export function a() {
  for (;;) {}
  return 1; // reported
}
export function b() {
  while (true) {}
  return 1; // not reported
}
export function c() {
  do {} while (true);
  return 1; // not reported
}
```

```js
export const o = {
  get a() {
    for (;;) {} // not reported
  },
  get b() {
    while (true) {} // reported: "This getter should return a value."
  },
};
```

`for (;;) {}` is treated as a loop that never exits, but `while (true) {}` and `do {} while (true)` are not, although both are equally infinite without a `break`. Two rules are affected:

| | `for (;;) {}` | `while (true) {}` | `do {} while (true)` |
| --- | --- | --- | --- |
| `noUnreachable`, code after the loop | reported | not reported | not reported |
| `useGetterReturn`, getter body is the loop | not reported | **reported** | not checked |

ESLint's `no-unreachable` reports the code after `while (true) {}`, and `getter-return` does not report the `while (true)` getter.

Reproduced with the 2.5.14 CLI and in the playground (f74eb38).

### Expected result

A loop whose condition is the literal `true` (or a truthy literal, as with `for (;;)`) and has no `break` targeting it is treated as never exiting. Code after it is unreachable, and a getter whose body is such a loop does not fall through.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

---

Tool Use Disclaimer: Claude Code using Anthropic Opus 5.5 helped build the reproduction.
